stdlib/SHA/src/types.jl

```diff
@@ -74,6 +89,20 @@ mutable struct SHA3_512_CTX <: SHA3_CTX
     bc::Array{UInt64,1}
 end
 
+function Base.getproperty(ctx::SHA3_CTX, fieldname::Symbol)
+    if fieldname === :state
+        return getfield(ctx, :state)::Vector{UInt64}
+    elseif fieldname === :bytecount
+        return getfield(ctx, :bytecount)::UInt128
+    elseif fieldname === :buffer
+        return getfield(ctx, :buffer)::Vector{UInt8}
+    elseif fieldname === :bc
+        return getfield(ctx, :bc)::Vector{UInt64}
+    else
+        error("SHA3_CTX has no field ", fieldname)
```

Comment:
If we want to exactly reproduce the normal error it would be

```suggestion
        error("type ", typeof(ctx), " has no field ", fieldname)
```

but perhaps it doesn't matter.